pkg/awsutil/awsutil.go

```diff
@@ -0,0 +1,36 @@
+package awsutil
+
+import (
+	"crypto/md5"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+	"math/big"
+	"reflect"
+	"strings"
+
+	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/crypto/ssh"
+)
+
+// https://tools.ietf.org/html/rfc4716#section-4
+func FingerprintPublicKey(key []byte) (string, error) {
+	sshPublicKey, _, _, _, err := ssh.ParseAuthorizedKey(key)
+	if err != nil {
+		return "", err
+	}
+	v := reflect.Indirect(reflect.ValueOf(sshPublicKey))
+	publicKey := &rsa.PublicKey{
+		N: v.FieldByName("N").Interface().(*big.Int),
+		E: v.FieldByName("E").Interface().(int),
+	}
```

Comment:
Actually, seeing as we always have the private key for this use case, we can just use that to get the public key in the form of an `rsa.PublicKey`.